src/window/concept_frame.rs

```diff
@@ -639,22 +641,20 @@ impl Frame for ConceptFrame {
                             // If theres no stored font data, find the first ttf regular sans font and
                             // store it
                             if self.font_data.is_none() {
-                                if let Some(font) = fontconfig::FontConfig::new()
-                                    .unwrap()
-                                    .get_regular_family_fonts(&font_face)
-                                    .unwrap()
-                                    .iter()
-                                    .find(|p| p.extension().map(|e| e == "ttf").unwrap_or(false))
-                                {
-                                    let mut font_data = Vec::new();
-                                    if let Ok(mut file) = ::std::fs::File::open(font) {
-                                        match file.read_to_end(&mut font_data) {
-                                            Ok(_) => self.font_data = Some(font_data),
-                                            Err(err) => {
-                                                log::error!("Could not read font file: {}", err)
-                                            }
-                                        }
-                                    }
+                                let font_bytes: Option<Vec<u8>> = fontconfig::FontConfig::new()
+                                    .and_then(|font_config: fontconfig::FontConfig| {
```

Comment:
Are the type annotations in these `and_then` calls necessary? I'd be surprised if they are.

Reply:
No, they aren't. I just included them to aid my understanding through the implementation. *Hopefully* this next commit will compile, and then it will be ready (but maybe need to squash b/c so many were failing)